README.md

```diff
@@ -29,4 +31,104 @@ Stage 1 of this bot will allow an authorized Discord member to:
 
 ### Future Stages
 
-To be determined!
\ No newline at end of file
+To be determined!
+
+## Usage
+
+This Discord bot allows you to run commands in a Discord channel like this:
+
+```
+~command
+```
+
+For example, if you run the ~ping command is a Discord channel, this bot will respond with "Pong!"
+
+```
+you: ~ping
+crates-io-bot: @you:Pong!
+```
+
+If you run the ~multiply command with two numbers, this bot will respond with the product of the two numbers.
+
+```
+you: ~multiply 2 2
+crates-io-bot: 4
+```
+
+These are obviously contrived commands, there will be commands specific to managing the crates.io infrastructure very soon.
+
+## Setup
+
+To setup this Discord bot, you need:
+* A [Discord Account](https://discordapp.com/))
+* A [Discord Server](https://support.discordapp.com/hc/en-us/articles/204849977-How-do-I-create-a-server-))
+* A [Heroku Account](https://www.heroku.com/)
+
+Go ahead and clone this repo and cd into the directory:
+
+```bash
+git clone https://github.com/rust-lang/crates-io-ops-bot.git
+cd crates-io-ops-bot
+```
+
+### Setting Up a Discord Application
+
+To use this bot, you will need to set up a Discord application through the [Discorse Developer Portal](https://discordapp.com/developers/). 
+
+[This blog post on Davao JS](https://medium.com/davao-js/2019-tutorial-creating-your-first-simple-discord-bot-47fc836a170b) has a good guide
+on creating a Discord application and generating the token key. Skip to the "Generating Token Key" heading in the post and come back here
+when you create the token.
+
+Make sure to store the token somewhere safe! You will need it!
+
+To use the token in development and testing, you need to add it to your .env file. 
+To set that file up:
+
+```bash
+cp .env.sample .env
+```
+
+Then add the token to your .env file:
+
+**.env**
+```
+DISCORD_TOKEN="<paste your token here>"
+```
+
+To use the token in production, make sure to set it wherever you define your environmental variables
+for your production environment.
+
+### Setting up Authorized Users
+
+Currently, all commands can be run by anyone in the Discord server running this bot.
+
+However, as we add new commands specific to managing infrastructure, some of those will be restricted to specific users.
+
+You can enable a user to run restricted commands by adding their Discord user name (not their server nickname) to the AUTHORIZED_USERS environmental variable.
```

Comment:
We probably want to gate on their Discord IDs, not on their usernames, as it's possible to change usernames. To make it simple a command like:

```
~myid
```

...could be useful (getting IDs from Discord needs the developer mode enabled).

Reply:
Done!